### PR TITLE
Sensing+

### PR DESCRIPTION
**Pull request by David-Orangemoon**
Sensing+ is an extension that expands the sensing part of scratch by adding blocks for mobile touch inputs and blocks allowing you to grab the lists and their content's from other sprites.
This block allows you to get the selected finger's X or Y position.
block_5_3_2023-10_16_37 PM
This block is a sprite exclusive block that allows you to get the finger that is currently touching that sprite or clone.
block_5_3_2023-10_16_32 PM
This block allows you to detect whether or not the sprite/clone is being touched by the specified finger.
block_5_3_2023-10_16_30 PM
This one returns true if the sprite/clone is touching any finger.
block_5_3_2023-10_16_27 PM
This one returns whether the specified finger is down.
block_5_3_2023-10_16_24 PM
Self explanitory.
block_5_3_2023-10_16_22 PM

Now onto the local list blocks
These are just ports of list blocks to allow for gathering local lists from other sprites.
block_5_3_2023-10_20_14 PM
block_5_3_2023-10_20_12 PM
block_5_3_2023-10_20_10 PM
block_5_3_2023-10_20_07 PM

The rest are really self explanitory
block_5_3_2023-10_21_31 PM
block_5_3_2023-10_21_28 PM
block_5_3_2023-10_21_26 PM
block_5_3_2023-10_21_25 PM

### CODE REVIEW

**Comment by David-Orangemoon** [comment]
The errors are from the device velocity readers.

**Comment by David-Orangemoon** [comment]
Sensing+
Its logo

**Comment by Xeltalliv** [comment]
Some suggestions:
* Currently starting and releasing touches shifts all touches after it, which makes it hard to determine which touch is which. It would be much easier to use if touches remained at their indices and touch list wasn't continuous.
* Maybe other than gyroscope and accelerometer also add [other sensors](https://github.com/TurboWarp/extensions/compare/master...Xeltalliv:extensions:generic-sensors)? (not sure) (some need to be enabled in chrome://flags)

**Comment by UndermanReal** [comment]
I think you should also add things like swiping your finger(up, down, left, right)

**Comment by David-Orangemoon** [comment]
> Some suggestions:
> * Currently starting and releasing touch shifts all touches after it, which makes it hard to determine which touch is which. It would be easier to use if touches remained at their indices and touch list wasn't continuous.
> * Maybe other than gyroscope and accelerometer also add [other sensors](https://github.com/TurboWarp/extensions/compare/master...Xeltalliv:extensions:generic-sensors)? (not sure) (some need to be enabled in chrome://flags)

I will fix the touch list bug. But with the sensors I still want to keep it somewhat scratchy so I think just grabbing the speed of the device will be good

**Comment by David-Orangemoon** [comment]
> I think you should also add things like swiping your finger(up, down, left, right)

And for these I will add swipe speed and direction

**Comment by David-Orangemoon** [comment]
> Some suggestions:
> 
> * Currently starting and releasing touch shifts all touches after it, which makes it hard to determine which touch is which. It would be easier to use if touches remained at their indices and touch list wasn't continuous.
> * Maybe other than gyroscope and accelerometer also add [other sensors](https://github.com/TurboWarp/extensions/compare/master...Xeltalliv:extensions:generic-sensors)? (not sure) (some need to be enabled in chrome://flags)

Ok I fixed the touches id thing.

**Comment by GarboMuffin** [comment]
The example file provided elsewhere did not work on this Android phone in Chrome/Brave

https://user-images.githubusercontent.com/33787854/236957544-e699917a-c7d9-4526-a2c7-fb79347aa027.mp4

**Comment by GarboMuffin** [comment]
Would it make sense for multitouch to be its own extension? Supposedly that's how another Scratch mod did theirs

**Comment by David-Orangemoon** [comment]
> 

give me a second I will check again.

**Comment by David-Orangemoon** [comment]
> The example file provided elsewhere did not work on this Android phone in Chrome/Brave
> 
>  screen-20230508-181714.2.mp4

It was a mistake with the project's code I think I fixed it 
https://drive.google.com/file/d/1x-gH34l-Mvk8NbbJpzFag2mN1Prv0yEn/view?usp=sharing

**Comment by David-Orangemoon** [comment]
block_5_8_2023-9_25_18 PM
block_5_8_2023-9_25_16 PM
block_5_8_2023-9_01_18 PM
block_5_8_2023-9_01_16 PM
Pain

**Comment by David-Orangemoon** [comment]
Pain 5 is just grammer

File: extensions/obviousAlexC/SensingPlus.js [hunk 1]
```diff
@@ -614,10 +634,39 @@
           toggleMenu: {
             items: ["on", "off"],
           },
+          effectMenu: {
```
Comment (GarboMuffin) [inline extensions/obviousAlexC/SensingPlus.js:668]: all of these need acceptReporters: true,

Comment (David-Orangemoon) [reply extensions/obviousAlexC/SensingPlus.js:668]: ok

**Comment by David-Orangemoon** [comment]
Fixed errors with the possibility to put anything into the effects.

**Comment by David-Orangemoon** [comment]
I'm finished for now I'm going to work in pen+

**Comment by David-Orangemoon** [comment]
I tried my best to resolve the conflicts but I can't figure it out. I don't use github that much.

**Comment by GarboMuffin** [comment]
do not worry about index.ejs merge conflicts. I resolve those at merge time

**Comment by David-Orangemoon** [comment]
> do not worry about index.ejs merge conflicts. I resolve those at merge time

Thanks man.

**Comment by David-Orangemoon** [comment]
Thanks garbo for refactoring alot of stuff!

**Comment by GarboMuffin** [comment]
status: the speech recognition blocks are the last thing that I still need to think about

**Comment by DNin01** [comment]
@GarboMuffin The clipboard blocks should probably ask for permission. You can block clipboard access with the browser if you want, but it'd also be nice to have a popup in TurboWarp like the one before fetching from URLs.

**Comment by GarboMuffin** [comment]
should probably do the same with speech recording too

**Pull request merged**